plugin.yaml

```diff
@@ -10,58 +10,215 @@ plugins:
     # URL to archive containing the plugin or name of directory containing
     # the plugin if it is included in the the blueprint directory under the
     # "plugins" directory.
-    source: https://github.com/cloudify-cosmo/cloudify-gcp-plugin/archive/master.zip
+    source: https://github.com/szpotona/cloudify-gcp-plugin/archive/add_keypair.zip
 
 node_types:
   cloudify.gcp.nodes.Instance:
     derived_from: cloudify.nodes.Compute
+    properties:
+      use_external_resource:
+        description: >
+          Indicate whether the resource exists or if Cloudify should create the resource.
+        type: boolean
+        default: false
+      gcp_config:
+        description: >
+          A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+        default: {}
+      image_id:
+        description: >
+          The ID of the image in your GCP account.
+        type: string
+      instance_type:
+        description: >
+          The instance's size.
+        type: string
+        default: n1-standard-1
+      properties:
+        description: >
+          Additional properties 
+        default: {}
     interfaces:
       cloudify.interfaces.lifecycle:
-        start:
+        create:
           implementation: gcp_plugin.plugin.tasks.create_instance
           inputs:
-            config:
-              default: {}
-        stop:
+            gcp_config:
+              default: { get_property: [SELF, gcp_config]}
+            properties:
+              default: { get_property: [SELF, properties]}
+            instance_type:
+              default: { get_property: [SELF, instance_type]}
+            image_id:
+              default: { get_property: [SELF, image_id]}
+        delete:
           implementation: gcp_plugin.plugin.tasks.delete_instance
           inputs:
-            config:
-              default: {}
+            gcp_config:
+              default: { get_property: [SELF, gcp_config] }
 
   cloudify.gcp.nodes.Network:
     derived_from: cloudify.nodes.Network
     properties:
-      network_name: 
-        description: The name of the network.
-        default: ''
+      use_external_resource:
+        description: >
+          Indicate whether the resource exists or if Cloudify should create the resource.
+        type: boolean
+        default: false
+      gcp_config:
+        description: >
+          A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+        default: {}
```

Comment:
the default should probably be { get_property: [SELF, gcp_config] } like you did in cloudify.gcp.nodes.Instance, no?